src/api/lib/authenticator.rb

```diff
@@ -87,11 +87,27 @@ def require_admin
     true
   end
 
+  def authorization_infos
+    # 1. try to get it where mod_rewrite might have put it
+    # 2. for Apache/mod_fastcgi with -pass-header Authorization
+    # 3. regular location
+    %w(X-HTTP_AUTHORIZATION Authorization HTTP_AUTHORIZATION).each do |header|
+      if request.env.has_key? header
+        return request.env[header].to_s.split
+      end
+    end
+    return
```

Comment:
There is an each iterator, so we need the return.

Probably would be nicer to return an array here. But I was just moving the code out of the private section. So I would prefer not to do that here.